### ISSUE TITLE
Error Message after converting SiteUrl

### ISSUE THREAD

**Issue by sawyerr44** [Zerg00s/FlowPowerAppsMigrator#2]
The following error message is displayed during a conversion:

Enter the URL of the original (old) SharePoint site: [OLD_SITE_URL]
WARNING: A newer version of PnP PowerShell is available: 3.23.2007.1. Consider upgrading.
resourceMapping.csv generated
Enter the URL of the destination SharePoint site: [NEW_SITE_URL]
resourceMapping.csv fully complete
Converting SiteUrl ...Done.
ForEach-Object : The term 'resources[$i].newId' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At C:\Development\Microsoft\FlowPowerAppsMigrator-master-mfa\ConvertPackage.ps1:20 char:14
+     $files | ForEach-Object {
+              ~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (resources[$i].newId:String) [ForEach-Object], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException,Microsoft.PowerShell.Commands.ForEachObjectCommand

I'm using MFA, so to get to this point I had to add -UseWebLogin after the site URL in your PowerShell scripts. I'm not sure if this has a knock on effect or it's something else?

I'am using the following versions:

Microsoft.Online.SharePoint.PowerShell 16.0.9021.0
SharePointPnPPowerShellOnline 3.23.2007.1
SharePointPnPPowerShellOnline 3.12.1908.1
SharePointPnPPowerShellOnline 3.6.1902.2

Your help would be much appreciated.

Regards

Rob

**Comment by Zerg00s** [comment]
Hi @sawyerr44,

I am not sure what is wrong yet, but I am happy to have a look at it if you can share more details:

- What are the contents of the `resourceMapping.csv` file? Perhaps it's empty for some reason.
- Are you moving MS Flow or Power App? 


As to MFA - I doubt it makes any difference.

**Comment by sawyerr44** [comment]
Hi,

Thank you for your response.

It did create the csv file and it also states to produce a file in the dest
folder.

I was moving a powerapp with connected flows.

I have been able to do it before without issue.

I did update to the latest SharePoint and pnp powershell beforehand.

Regards

Rob

**Comment by Zerg00s** [comment]
hi @sawyerr44 ,

I have tried running the script with SharePointPnPPowerShellOnline version **3.23.2007.1** and with Connect-PnPOnline **-UseWebLogin**. I didn't get any errors. 

In ConvertPackage.ps1 file, lines 20 to 30: could you double-check that you are not missing **$** character before  `resources[$i].newId` ? I think, this is the most likely explanation to the error you are getting.

**Comment by sawyerr44** [comment]
Hi @Zerg00s ,

I have checked the code and everything looks fine. The Resources and IDs have a $ in front, except for the entry in the Write.host.

I think it's getting further along the process as files are being produced in the dest folder. The files are not zipped though as in your screen shots.

The resource mappings spreadsheet also contains 31 entries. The newID column is missing some entries (8 out of the 31). Could this possibly cause the error?

Some facts about my solution that may help the investigation:

- I'm not moving the solution across tenants, but in the same tenant. (Different site collections)
- Another two SharePoint lists (in the same site collection) are referenced in the PowerApp (these are missing the newID)
- The Flows are triggered from within the PowerApps via buttons.

In the meantime, I will download your latest package and retry.

Regards

Rob

**Comment by sawyerr44** [comment]
Hi @Zerg00s ,

I wanted to give you an update. I have been able to complete a successful run. Here are the steps that I implemented:

- I reinstalled the SharePoint PowerShell module.
- Downloaded your updated files.
- Added my files to convert and run the conversation.
- It ran successfully!

The only error I ran into which caught me out the first time I ran your tool is that the RunAllScripts.ps1 filename is missing the "c" in scripts.

Thank you for your help.

Regards

Rob
